tests/unit/test_history_injection.py

```
@@ -0,0 +1,280 @@
+    history = [
+        {"role": "system", "content": "Existing system message"},
+        {"role": "user", "content": "Hello"}
+    ]
+    mock_agent._run_init(task="New task", chat_history=history)
+    
+    from unittest.mock import MagicMock, AsyncMock
+    mock_agent.planner = MagicMock()
+    mock_agent.planner.get_formatted_plan.return_value = "Plan"
+    mock_agent.planner.create_plan = AsyncMock()
+    
+    async for _ in mock_agent.pre_run(): pass
```

Ruff (0.15.12)

[error] Multiple statements on one line (colon)

(E701)